src/wombats/game/processor.clj

```diff
@@ -103,23 +103,106 @@
     (.setPayload request payload)
     request))
 
+(defn- sh-handler
+  "This runs some code by creating a script and running it against an
+  interpreter. It feeds it the state on STDIN and reads the next move
+  from STDOUT.
+
+  If the subprocess throws an error, that's translated into an
+  exception."
+  [ext interpreter state code]
+  (let [tmp-file (java.io.File/createTempFile "bot" ext)
+        input (cheshire/generate-string state)]
+    (try
+      (with-open [file (clojure.java.io/writer tmp-file)]
+        (.write file code))
+      (let [proc (clojure.java.shell/sh interpreter (.getAbsolutePath tmp-file)
+                                        :in (java.io.StringReader. input))]
+        (if (zero? (:exit proc))
+          (cheshire/parse-string (:out proc) true)
+          (throw (Exception. (:err proc)))))
+      (finally
+        (.delete tmp-file)))))
+
+(defn- js-handler
+  "This evaluates the js bot code against the state given."
+  [code state timeout]
+  (sh-handler ".js" "node" state
+        (str "const stdin  = process.stdin,\n"
+             "      stdout = process.stdout;\n\n"
+             "const wombats = "
+             code
+             "\n\nconst timeLeft = " timeout "\n\n"
+             "var chunks = [];\n\n"
+             "stdin.resume();\n"
+             "stdin.setEncoding('utf8');\n\n"
+             "stdin.on('data', function(chunk) {\n"
+             "  chunks.push(chunk);\n"
+             "});\n\n"
+             "stdin.on('end', function() {\n"
+             "  var inputJSON  = chunks.join(),\n"
+             "      parsedData = JSON.parse(inputJSON),\n"
+             "      outputData = wombats(parsedData, timeLeft);\n\n"
+             "  stdout.write(JSON.stringify(outputData, null, '  '));\n"
+             "});\n")))
+
+(defn- clj-handler
+  "This evaluates the clj bot code against the state given."
+  [code state timeout]
```

Comment:
this may get a bit more complicated, but `timeout` for each language is a function to be called that will return in milliseconds how much time is remaining before the user is cutoff.